src/Hooks/search/createWfsSearchFunction.ts

```diff
@@ -57,17 +93,25 @@ export const createWfsSearchFunction = <
         return response.json() as unknown as C;
       } else {
         const responseString = await response.text();
+        const formatGeoJson = new OlFormatGeoJSON();
 
-        const formatGml = new OlFormatGml3({
-          featureNS,
-          srsName
-        });
+        if (gmlVersion) {
+          const formatGml = createGmlFormat(gmlVersion, featureNS, srsName);
+          return formatGeoJson.writeFeaturesObject(
+            formatGml.readFeatures(responseString)
+          ) as C;
+        }
 
-        const formatGeoJson = new OlFormatGeoJSON();
-        return formatGeoJson.writeFeaturesObject(formatGml.readFeatures(responseString)) as C;
+        const gmlVersions: GmlVersion[] = ['GML32', 'GML3', 'GML2'];
+        for (const version of gmlVersions) {
+          const formatGml = createGmlFormat(version, featureNS, srsName);
+          const features = formatGml.readFeatures(responseString);
+          if (features.length > 0) {
```

Comment:
What happens if the source is actually empty? Probably it's less error prone if we wrap the parsing in a try catch?

Reply:
Yes, thanks! I wrapped the parsing logic in a `try-catch` block to handle potential errors when reading features from the WFS response.  
If parsing fails for one GML version, the next one will be attempted without interrupting the entire process.

Errors are logged as warnings.